tools/windows-node-installer/pkg/cloudprovider/factory.go

```diff
@@ -4,54 +4,35 @@ import (
 	"fmt"
 
 	v1 "github.com/openshift/api/config/v1"
-	clientset "github.com/openshift/client-go/config/clientset/versioned"
 	"github.com/openshift/windows-machine-config-operator/tools/windows-node-installer/pkg/client"
 	"github.com/openshift/windows-machine-config-operator/tools/windows-node-installer/pkg/cloudprovider/aws"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// CloudProviderFactory returns a cloud specific interface
-// The factory takes in kubeconfig of an existing OpenShift Cluster and Cloud vendor specific credential file
-// It also requires name of the credential account since there might be more than one accounts in the file
-// Also a directory which should be default to the current working directory where install information
-// such as created instance and security group id should be saved in.
-// The returned cloud specific interface will be able to perform rest of the necessary functions related to creating or
-// destroying an instance
-func CloudProviderFactory(kubeconfigPath, credentialPath, credAccount, dir *string) (Cloud, error) {
+// CloudProviderFactory returns a cloud specific interface for future interactions.
+// The factory takes in kubeconfig of an existing OpenShift cluster and a cloud vendor specific credential file.
+// Since the credential file may contain multiple accounts and the default account name/ID varies between providers,
+// this function requires specifying the credentialAccountID of the user's credential account.
+// The resourceTrackerDir is where the 'installed information' such as created instance and security group ID's is
+// saved.
+// The function returns cloud specific interface for performing necessary functions related to creating or destroying
+// an instance
+func CloudProviderFactory(kubeconfigPath, credentialPath, credentialAccountID, resourceTrackerDir string) (Cloud, error) {
 	// Create a new client of the given OpenShift cluster based on kubeconfig
-	kubeclient, err := client.New(kubeconfigPath)
+	oc, err := client.GetOpenShiftClient(kubeconfigPath)
 	if err != nil {
 		return nil, err
 	}
-	// Obtain cluster infrastructure information which includes Name and Platform
-	infra, err := GetInfrastructure(kubeclient)
+	// Obtain cluster infrastructure information which includes name and provider
```

Comment:
Please be consistent. Why is this `cluster infrastructure` in some places and `cloud provider` in others? Please stick with one or the other?